portfolio/src/main/webapp/script-map.js

```diff
@@ -0,0 +1,144 @@
+  });
+
+  createMarkerForDisplay(
+      39.329858, -76.620540,
+      'I go to school here.', 'Johns Hopkins University', false)
+  createMarkerForDisplay(
+      32.003407, 118.734956,
+      'This is where I was born.', 'Nanjing', false)
+  createMarkerForDisplay(
+      37.423829, -122.092154,
+      'This is where my internship was supposed to be, but now it\'s remote.', 'Google', false);
+  createMarkerForDisplay(
+      23.117487, -82.373375,
+      'I went there with my friends. It was great fun.', 'Havana', false);
+  createMarkerForDisplay(
+      38.642805, -90.195679,
+      'I worked here during summer, 2019.', 'St Louis', false);
+
+  fetchMarkers();
+}
+
+/** 
+ * Adds a marker that shows an info window when clicked. 
+ * @param {number} lat Latitude of the marker position.
+ * @param {number} lng Longitude of the marker position.
+ * @param {string} content The content of the marker, which is displayed onclick.
+ * @param {string=} title Optional title of the marker, which is displayed on hover.
+ * @param {boolean=} defaultIcon Whether to display the default marker icon or the flagIcon. 
+ */
+function createMarkerForDisplay(lat, lng, content, title = '', defaultIcon = true) {
+  let marker;
+  if (defaultIcon == false) {
+    marker = new google.maps.Marker(
+        {position: {lat: lat, lng: lng}, map: map, title: title, icon: flagIcon});
+  } else {
+    marker = new google.maps.Marker(
+      {position: {lat: lat, lng: lng}, map: map});
+  }
+
+  const infoWindow = new google.maps.InfoWindow({content: content});
+  marker.addListener('click', () => {
+    infoWindow.open(map, marker);
+  });
+}
+
+/** Fetches markers from the backend and adds them to the map. */
+function fetchMarkers() {
+  fetch('/markers').then(response => response.json()).then((markers) => {
+    markers.forEach(
+        (marker) => {
+            createMarkerForDisplay(marker.lat, marker.lng, marker.content)});
+  });
+}
+
+/** 
+ * Creates a marker that shows a textbox the user can edit. 
+ * @param {number} lat Latitude of the marker position.
+ * @param {number} lng Longitude of the marker position.
+ */
+function createMarkerForEdit(lat, lng) {
+  // If we're already showing an editable marker, then remove it.
+  if (editMarker) {
+    editMarker.setMap(null);
+  }
+
+  editMarker = new google.maps.Marker({position: {lat: lat, lng: lng}, map: map});
+
+  const infoWindow = new google.maps.InfoWindow({content: buildInfoWindowInput(lat, lng)});
+
+  // When the user closes the editable info window, remove the marker.
+  google.maps.event.addListener(infoWindow, 'closeclick', () => {
+    editMarker.setMap(null);
+  });
+
+  infoWindow.open(map, editMarker);
+}
+
+/** 
+ * Builds and returns HTML elements that show an editable textbox and a submit button. 
+ * @param {number} lat Latitude of the marker with the infoWindow.
+ * @param {number} lng Longitude of the marker with the infoWindow.
```

Comment:
If I'm understanding right, the marker has the info window?


```suggestion
 * @param {number} lat Latitude of the marker containing the infoWindow.
 * @param {number} lng Longitude of the marker containing the infoWindow.
```

Reply:
Well, the info window will pop up when the marker is clicked, so I wouldn't say that it is contained

Reply:
Would you read it as:
(latitude of the marker) with the window
latitude of the (marker with the window)

if its the second one where the window is associated with the marker, and not its own entity, say something like:

latitude of the marker having a popup window or something thing like that

Reply:
Ah didn't realize the ambiguity lol. I'll say the marker associated with the pop up window.